packages/dart_frog_test/lib/src/dart_frog_test.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:async';
+
+import 'package:dart_frog/dart_frog.dart';
+import 'package:meta/meta.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:test/test.dart';
+
+class _MockRequestContext extends Mock implements RequestContext {}
+
+/// {@template test_request}
+/// A model used to create a dart frog [Request] in a test.
+/// {@endtemplate}
+class TestRequest {
+  /// {@macro test_request}
+  TestRequest({
```

Comment:
Note: Ideally this class wouldn't exist, but Dart frog's `Request` class isn't super API friendly for tests since you:

(a) have to provide the method in string format, which is way less prone to typos than using `HttpMethod.x`
(b) you have to provide an `Uri`, which can be cumbersome to create since you need to provide the full path in order to use `Uri.parse`

In `TestRequest` use use the http method enum instead, making the test type safer and ask only for the path from the user, using a placeholder base url to make the tester easier to write and read.

Reply:
For (a) users could use the other constructors we provide as Request.get and so on, right?

Reply:
> For (a) users could use the other constructors we provide as Request.get and so on, right?

yes! but the default constructor is still available to them, so they could use and run into a confusing error